### PR TITLE
Fix TLV tags in the Onboarding payload and update unit tests

### LINKED ISSUE #10187: TLV tag use in SetupPayload does not seem to match spec
#### Problem
In SetupPayload.cpp we have:
```
bool IsCHIPTag(uint8_t tag)
{
    return tag >= (1 << kRawVendorTagLengthInBits);
}

bool IsVendorTag(uint8_t tag)
{
   return tag < (1 << kRawVendorTagLengthInBits);
}
```

but in the spec "5.1.4.1. Manufacturer-specific Elements" says:

> Manufacturer-specific elements SHALL be tagged with context-specific tags that have semantics which are defined by the vendor for use in the products using their Vendor ID, and SHALL use tag values 0x80 to 0xFF.  Tag values 0x00 to 0x7F are reserved to indicate CHIP-common elements.
    
which seems exactly backwards from the code.

What exactly is going on here?  @shana-apple 

 #### Proposed Solution
Sort out what's going on, and fix or document to make it clear how this code and the spec interact.  Add tests as needed.

### PR DESCRIPTION
**Pull request by harsha-rajendran**
#### Problem
Fixes #10187 

#### Change overview
Update the Manufacture-specific and CHIPCommon optional tags in the Onboarding Payload. Updated unit tests to reflect this change.

#### Testing
- Updated Unit tests
- Sanity-check: Ran OTA test to ensure commissioning and OTA transfer is unaffected

### CODE REVIEW

File: src/setup_payload/SetupPayload.cpp [hunk 1]
```diff
@@ -32,14 +32,16 @@
 
 namespace chip {
 
-bool IsCHIPTag(uint8_t tag)
+// Spec 5.1.4.2 CHIPCommon tag numbers are in the range [0x00, 0x7F]
+bool IsCommonTag(uint8_t tag)
 {
-    return tag >= (1 << kRawVendorTagLengthInBits);
+    return tag < (1 << kRawVendorTagLengthInBits);
```
Comment (bzbarsky-apple) [inline src/setup_payload/SetupPayload.cpp:38]:
```suggestion
    return tag < 0x80;
```
I don't think `kRawVendorTagLengthInBits` makes sense in this new setup as a concept.

File: src/setup_payload/SetupPayload.cpp [hunk 2]
```diff
@@ -32,14 +32,16 @@
...
 }
 
+// Spec 5.1.4.1 Manufacture-specific tag numbers are in the range [0x80, 0xFF]
 bool IsVendorTag(uint8_t tag)
 {
-    return tag < (1 << kRawVendorTagLengthInBits);
+    return tag >= (1 << kRawVendorTagLengthInBits);
```
Comment (bzbarsky-apple) [inline src/setup_payload/SetupPayload.cpp:44]:
```suggestion
    return tag >= 0x80;
```
Or maybe just `!IsCommonTag`.

File: src/setup_payload/SetupPayload.h [hunk 3]
```diff
@@ -160,7 +161,7 @@ struct OptionalQRCodeInfoExtension : OptionalQRCodeInfo
     uint64_t uint64;
 };
 
-bool IsCHIPTag(uint8_t tag);
+bool IsCommonTag(uint8_t tag);
 bool IsVendorTag(uint8_t tag);
```
Comment (bzbarsky-apple) [inline src/setup_payload/SetupPayload.h:165]: Could we make these static SetupPayload methods instead of global ones?

**Review by bzbarsky-apple — APPROVED** [review]
Thank you!

File: src/setup_payload/SetupPayload.cpp (hunk 2 above)
Comment (hrajendran-samsung) [reply src/setup_payload/SetupPayload.cpp:44]: Good point

File: src/setup_payload/SetupPayload.h (hunk 3 above)
Comment (hrajendran-samsung) [reply src/setup_payload/SetupPayload.h:165]: IsCommonTag() is being used in another compilation unit (QRCodeSetupPayloadParser.cpp), unfortunately.

Comment (bzbarsky-apple) [reply src/setup_payload/SetupPayload.h:165]: Yes, but if it's a class static method that's not a problem.

Comment (hrajendran-samsung) [reply src/setup_payload/SetupPayload.h:165]: Ah, put it in the SetupPayload class!?. Got it.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**